### ISSUE TITLE
eslint --init fails when reviewing files using async functions

### ISSUE THREAD

**Issue by benjaminapetersen** [eslint/eslint#10045]
**Tell us about your environment**

* **ESLint Version:** v4.18.1
* **Node Version:** v9.4.0
* **npm  Version:** 5.6.0
* **yarn  Version:** v0.19.1 (using instead of npm)

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

n/a, bug is the "create configuration" `--init` process

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Here is the output of following the CLI menus, as well as the error:

```bash 
$ $(yarn bin)/eslint --init
? How would you like to configure ESLint? Inspect your JavaScript file(s)
? Which file(s), path(s), or glob(s) should be examined? puppeteer.test.js
? What format do you want your config file to be in? YAML
? Are you using ECMAScript 6 features? Yes
? Are you using ES6 modules? No
? Where will your code run? Node
? Do you use JSX? No

(~/projects/my-project/puppeteer.test.js:3:11) Parsing error: Unexpected token =>
Error: (~/projects/my-project/puppeteer.test.js:3:11) Parsing error: Unexpected token =>
    at getSourceCodeOfFile (~/projects/my-project/node_modules/eslint/lib/util/source-code-util.js:38:15)
    at filenames.forEach.filename (~/projects/my-project/node_modules/eslint/lib/util/source-code-util.js:94:28)
    at Array.forEach (<anonymous>)
    at getSourceCodeOfFiles (~/projects/my-project/node_modules/eslint/lib/util/source-code-util.js:93:15)
    at configureRules (~/projects/my-project/node_modules/eslint/lib/config/config-initializer.js:165:23)
    at processAnswers (~/projects/my-project/node_modules/eslint/lib/config/config-initializer.js:288:18)
    at inquirer.prompt.then.secondAnswers (~/projects/my-project/node_modules/eslint/lib/config/config-initializer.js:536:32)
    at <anonymous>
Determining Config: 0% [------------------------------] 0.0s elapsed, eta 0.0s
```

```js
// ran against this sample code provided for trying out puppeteer
const puppeteer = require('puppeteer');

(async () => {
  const browser = await puppeteer.launch();
  const page = await browser.newPage();
  await page.goto('https://example.com');
  await page.screenshot({path: 'example.png'});

  await browser.close();
})();

```

```bash

```

**What did you expect to happen?**

A generated config allowing the use of `es8 async functions`.

**What actually happened? Please include the actual, raw output from ESLint.**

The error above.

Thanks!

**Comment by lamflam** [comment]
I think this is the same issue as #10003

**Comment by ilyavolodin** [comment]
Yes, this is a duplicate of #10003 When `init` command was created, ES6 was the latest version, and we haven't updated it since. Closing as a duplicate.
